src/model.js

```diff
@@ -55,7 +55,12 @@ const xget = async (req, res, next) => {
 				.limit(limit)
 				.sort(sort)
 				.exec();
-			res.json(records.map(r => helper.responseBody(r, req.collection)));
+			
+			if (req.headers['x-get-metadata']) {
+				res.json(helper.getRecordsMetadata(records));
```

Comment:
@harlev The records in this line is actually the filtered result and not the whole record list. For eg., if we pass a `q` param this result will have records only satisfying to the query.

Reply:
Yes. now separated to a different route and using the basic list of records